mm2src/mm2_main/src/lp_wallet/mnemonics_wasm_db.rs

```diff
@@ -0,0 +1,146 @@
+use crypto::EncryptedMnemonicData;
+use mm2_core::mm_ctx::MmArc;
+use mm2_core::DbNamespaceId;
+use mm2_db::indexed_db::{DbIdentifier, DbInstance, DbUpgrader, IndexedDb, IndexedDbBuilder, OnUpgradeResult,
+                         TableSignature};
+use mm2_err_handle::prelude::*;
+use std::collections::HashMap;
+
+type WalletsDBResult<T> = Result<T, MmError<WalletsDBError>>;
+
+#[derive(Debug, Deserialize, Display, Serialize)]
+pub enum WalletsDBError {
+    #[display(fmt = "Error deserializing '{}': {}", field, error)]
+    DeserializationError {
+        field: String,
+        error: String,
+    },
+    #[display(fmt = "Error serializing '{}': {}", field, error)]
+    SerializationError {
+        field: String,
+        error: String,
+    },
+    Internal(String),
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+struct MnemonicsTable {
+    wallet_name: String,
+    encrypted_mnemonic: String,
+}
+
+impl TableSignature for MnemonicsTable {
+    fn table_name() -> &'static str { "mnemonics" }
+
+    fn on_upgrade_needed(upgrader: &DbUpgrader, old_version: u32, new_version: u32) -> OnUpgradeResult<()> {
+        if let (0, 1) = (old_version, new_version) {
+            let table = upgrader.create_table(Self::table_name())?;
+            table.create_index("wallet_name", true)?;
+        }
+        Ok(())
```

Comment:
I think it's better to use an approach similar to 
https://github.com/KomodoPlatform/komodo-defi-framework/blob/89f5ffa903c01892ee86e3b80c669503f4a23e44/mm2src/mm2_main/src/lp_swap/swap_wasm_db.rs#L167-L190

E.g., if we need to migrate from 0 to 2, the current implementation will skip `create_table` step. Or we will need to handle all cases: `(0, 1)`, `(0, 2)`, `(1, 2)`, which can become bulky with possible future version upgrades.

Reply:
That's a good approach, will follow it. Thanks.

Reply:
done